test/extended/builds/pipeline_origin_bld.go

```diff
@@ -274,33 +274,41 @@ var _ = g.Describe("[Feature:Builds][Feature:Jenkins][Slow] openshift pipeline b
 			})
 
 			g.By("should verify services successfully", func() {
-				// instantiate the bc
-				g.By("create the jenkins pipeline strategy build config that leverages openshift client plugin")
-				err := oc.Run("new-app").Args("-f", verifyServiceClientPluginPipelinePath).Execute()
-				o.Expect(err).NotTo(o.HaveOccurred())
 
-				err = exutil.WaitForDeploymentConfig(oc.KubeClient(), oc.AppsClient().AppsV1(), oc.Namespace(), "nodejs-example", 1, false, oc)
+				newTestAppArgs := []string{"--template", "redis-ephemeral", "-p", "MEMORY_LIMIT=128Mi"}
+
+				// Redis deployment with the redis service
+				g.By("instantiate the test application")
+				err := oc.Run("new-app").Args(newTestAppArgs...).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				err = exutil.WaitForDeploymentConfig(oc.KubeClient(), oc.AppsClient().AppsV1(), oc.Namespace(), "redis", 1, false, oc)
 				if err != nil {
-					exutil.DumpApplicationPodLogs("nodejs-example", oc)
+					exutil.DumpApplicationPodLogs("redis", oc)
 				}
 
+				// Redis headless service and jenkinsFile which runs verify service on both the services
+				g.By("create the jenkins pipeline strategy build config that leverages openshift client plugin")
+				err = oc.Run("new-app").Args("-f", verifyServiceClientPluginPipelinePath).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
 				brservices, err := exutil.StartBuildAndWait(oc, "jenkins-verifyservice-pipeline")
 				if err != nil || !brservices.BuildSuccess {
 					debugAnyJenkinsFailure(brservices, oc.Namespace()+"-jenkins-verifyservice-pipeline", oc, true)
 					exutil.DumpBuilds(oc)
-					exutil.DumpDeploymentLogs("nodejs-example", 1, oc)
+					exutil.DumpDeploymentLogs("redis", 1, oc)
 					exutil.DumpBuildLogs("jenkins-verifyservice-pipeline", oc)
 				}
 				brservices.AssertSuccess()
+
 				g.By("get build console logs and see if succeeded")
 				_, err = j.GetJobConsoleLogsAndMatchViaBuildResult(brservices, "Finished: SUCCESS")
 				o.Expect(err).NotTo(o.HaveOccurred())
 				g.By("clean up openshift resources for next potential run")
 				err = oc.Run("delete").Args("bc", "jenkins-verifyservice-pipeline").Execute()
 				o.Expect(err).NotTo(o.HaveOccurred())
-				err = oc.Run("delete").Args("dc", "nodejs-example").Execute()
+				err = oc.Run("delete").Args("dc", "redis").Execute()
 				o.Expect(err).NotTo(o.HaveOccurred())
-				err = oc.AsAdmin().Run("delete").Args("all", "-l", "app=nodejs-example").Execute()
+				err = oc.AsAdmin().Run("delete").Args("all", "-l", "app=redis-ephemeral").Execute()
```

Comment:
this one needs to be fixed: Change to app=redis or better to "app=" + redisAppName .

also, note that local imageStreams in the namespace are not holding the level.
so, maybe add an additional `oc delete is redis` or `oc delete is --all`.